flang/test/Lower/global-init.f90

```diff
@@ -21,12 +21,16 @@ subroutine foo2()
 end subroutine
 subroutine foo3()
```

Comment:
I think you might want to use `CHECK-LABEL` to anchor the `address_of` ops against the `foo3` code block. You might also want to use `CHECK-DAG` to allow these ops to come out in a different order.

Reply:
I'll give that a shot.